tests/sentry/tasks/test_commit_context.py

```diff
@@ -534,6 +535,55 @@ def test_gh_comment_no_pr(self, get_jwt, mock_comment_workflow):
             )
             assert not mock_comment_workflow.called
 
+    @with_feature("organizations:pr-comment-bot")
+    @patch("sentry.integrations.github.client.get_jwt", return_value=b"jwt_token_1")
+    @responses.activate
+    def test_gh_comment_commit_not_in_default_branch(self, get_jwt, mock_comment_workflow):
+        """No comments on commit not in default branch"""
+
+        responses.add(
+            responses.POST,
+            self.base_url + f"/app/installations/{self.installation_id}/access_tokens",
+            json={"token": self.access_token, "expires_at": self.expires_at},
+        )
+        responses.add(
+            responses.GET,
+            self.base_url + f"/repos/example/commits/{self.commit.key}/pulls",
+            status=200,
+            json=[{"merge_commit_sha": "abcd"}, {"merge_commit_sha": "efgh"}],
+        )
+
+        with self.tasks():
+            event_frames = get_frame_paths(self.event)
+            process_commit_context(
+                event_id=self.event.event_id,
+                event_platform=self.event.platform,
+                event_frames=event_frames,
+                group_id=self.event.group_id,
+                project_id=self.event.project_id,
+            )
+            assert not mock_comment_workflow.called
+
+    @with_feature("organizations:pr-comment-bot")
+    @patch("sentry.integrations.github.client.get_jwt", return_value=b"jwt_token_1")
+    @responses.activate
+    def test_gh_comment_no_pr_from_query(self, get_jwt, mock_comment_workflow):
+        """No comments on suspect commit with no pr"""
```

Comment:
Could we make this distinct from the docstring on line 512? Also is there a reason this test is separated from the `from_api` version or can we move them together?

Reply:
in the `from_api` test, the mocked API response returns no associated pull request so we have an early return in the check on line 51 in `commit_context.py`. in this test, `pr_query` does not exist so we return early in the check on line 69